src/utils/rest-api.js

```diff
@@ -422,11 +422,10 @@ export function elementExists(directoryUuid, elementName, type) {
     console.debug(existsElementUrl);
     return backendFetch(existsElementUrl, { method: 'head' }).then(
         (response) => {
-            return response.ok
-                ? true
-                : response.status === 404
-                ? false
-                : Promise.reject(response.statusText);
+            if (response.ok) {
+                return response.status !== 204; // HTTP 204 : No-content
+            }
+            return Promise.reject(response.statusText);
```

Comment:
return response.ok ?
? response.status !== 204
: response.text().then((text) => Promise.reject(text)) :

Reply:
I don't like mixing ternary operators and multiple return types. When reading the code, it can be easy to overlook the fact that we could return three different things : true, false, and reject.

I think it is easier to read something like this : 
`return condition ? value : value;`
than something like this : 
`return condition ? conditionalValue : value;`
In the second example, it is easy to overlook the `conditionalValue` and not realize it could lead to multiple and different return values.

By using a simple if clause, it is easier to differenciate between the conditional returns and the reject one.

Reply:
Right